iceoryx_examples/icediscovery_in_c/README.md

```diff
@@ -0,0 +1,131 @@
+# icediscovery in C
+
+## Introduction
+
+This example demonstrates how to search for specific services using iceoryx's
+service discovery. It provides two applications - one offering different
+services and one searching for these making different search queries. The
+behavior and structure is quite similar to the [icediscovery C++ example](https://github.com/eclipse-iceoryx/iceoryx/tree/v2.0.0/iceoryx_examples/icedelivery).
+
+<!--## Expected Output-->
+<!-- @todo Add expected output with asciinema recording before v2.0-->
+
+## Code walkthrough
+
+### Offer services
+
+We create several publisher ports which offer their services on construction by
+default. For more dynamism the `cameraPublishers` offer/stop their services
+periodically. If you want more information on how to create publisher ports,
+have a look at the [icedelivery C example](https://github.com/eclipse-iceoryx/iceoryx/tree/v2.0.0/iceoryx_examples/icedelivery_in_c).
+
+### Find services
+
+To be able to search for services, we have to create a service discovery handle
+which points to `storage`, the memory location where the service discovery is
+stored.
+
+<!--[geoffrey][iceoryx_examples/icediscovery_in_c/iox_c_find_service.c][create service discovery handle]-->
+```c
+iox_service_discovery_storage_t storage;
+iox_service_discovery_t serviceDiscovery = iox_service_discovery_init(&storage);
+```
+
+It is included via:
+
+<!--[geoffrey][iceoryx_examples/icediscovery_in_c/iox_c_find_service.c][include service discovery]-->
+```c
+#include "iceoryx_binding_c/service_discovery.h"
+```
+
+We can now call three different find service functions. Let's start with
+
+<!--[geoffrey][iceoryx_examples/icediscovery_in_c/iox_c_find_service.c][find service and apply callable]-->
+```c
+iox_service_discovery_find_service_apply_callable(
```

Comment:
The terminology of callable is more of a C++ one, in C only functions and function pointers are callable and we can only pass the latter (there are no C++ functors). Still, I would keep the name as it is correct but maybe unusual to a C user.